sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ExtRefService.java

```diff
@@ -196,43 +191,170 @@ public static List<TExtRef> filterDuplicatedExtRefs(List<TExtRef> tExtRefs) {
+                && extRef.getDesc().endsWith("_" + tChannel.getDAName() + "_1");
+        Boolean isExtRefDescMatchDigitalChannel = tChannel.getChannelType().equals(TChannelType.DIGITAL)
+                && extRef.getDesc().startsWith("DYN_LDEPF_DIGITAL CHANNEL " + tChannel.getChannelNum()+"_1_BOOLEEN")
+                && extRef.getDesc().endsWith("_" + tChannel.getDAName() + "_1");
+        return extRef.isSetDesc() && (isExtRefDescMatchAnalogChannel || isExtRefDescMatchDigitalChannel)
+                && extRef.isSetPLN() && Utils.lnClassEquals(extRef.getPLN(), tChannel.getLNClass())
+                && extRef.isSetPDO() && extRef.getPDO().equals(tChannel.getDOName());
+    }
+
+    /**
+     * Verify whether the IED satisfies the EPF channel for the private element `TCompasICDHeader`
+     * @param iedAdapter IEDAdapter
+     * @param channel TChannel
+     * @return true if the TCompasICDHeader matches the EPF channel
+     */
+    private static boolean isIcdHeaderMatchLDEPFChannel(IEDAdapter iedAdapter, TChannel channel) {
+        return iedAdapter.getCompasICDHeader()
+                .map(compasICDHeader -> compasICDHeader.getIEDType().value().equals(channel.getIEDType())
+                        && compasICDHeader.getIEDredundancy().value().equals(channel.getIEDRedundancy().value())
+                        && compasICDHeader.getIEDSystemVersioninstance().toString().equals(channel.getIEDSystemVersionInstance()))
+                .orElse(false);
+    }
+
+    /**
+     * Provides Active LDevice according to EPF channel's inst attribute
+     * @param iedAdapter IEDAdapter
+     * @param channel TChannel
+     * @return LDeviceAdapter object that matches the EPF channel
+     */
+    private static Optional<LDeviceAdapter> getActiveSourceLDeviceByLDEPFChannel(IEDAdapter iedAdapter, TChannel channel) {
+        return iedAdapter.findLDeviceAdapterByLdInst(channel.getLDInst())
+                .filter(lDeviceAdapter -> lDeviceAdapter.getLDeviceStatus()
+                        .map(status -> status.equals(LdeviceStatus.ON.getValue()))
+                        .orElse(false))
+                .stream().findFirst();
+    }
+
+    /**
+     * Provides Active LN Object that satisfies the EPF channel attributes (lnClass, lnInst, prefix)
+     * @param lDeviceAdapter LDeviceAdapter
+     * @param channel TChannel
+     * @return AbstractLNAdapter object that matches the EPF channel
+     */
+    private static Optional<AbstractLNAdapter<?>> getActiveLNodeSourceByLDEPFChannel(LDeviceAdapter lDeviceAdapter, TChannel channel) {
+        return lDeviceAdapter.getLNAdaptersIncludingLN0().stream()
+                .filter(lnAdapter -> lnAdapter.getLNClass().equals(channel.getLNClass())
+                        && lnAdapter.getLNInst().equals(channel.getLNInst())
+                        && Utils.equalsOrBothBlank(channel.getLNPrefix(), lnAdapter.getPrefix()))
```

Comment:
`equalsOrBothBlank` has more than 30 usages, so it would be better to create another issue for this.
I also see the rconf project using this.
@massifben Could you open a ticket noting it as deprecated ?

Reply:
equalsOrBothBlank is often used to compare LN key attributes (lnClass, lnInst, prefix).
I created a ticket in RCONF to migrate LnKey record to Compas and refacto Compas. We will mark equalsOrBothBlankas deprecated then. I hava updated the RCONF ticket RSR-891

Reply:
Thanks, I only applied @gleizesDor suggestion without deprecate `equalsOrBothBlank`.